cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -528,6 +529,22 @@ ClusterOperationMetadata createCluster(String name, String imageVersion, Map<Str
     }
   }
 
+  private void cleanUpClusterAfterCreationFailure(String name) {
+    if (conf.isSkipDelete()) {
+      //Don't delete even failed one when skip delete is set
+      return;
+    }
+    try {
+      Optional<Cluster> cluster = getDataprocCluster(name)
+        .filter(c -> c.getStatus().getState() == ClusterStatus.State.ERROR);
+      if (cluster.isPresent()) {
+        deleteCluster(name);
+      }
+    } catch (Exception e) {
+      LOG.warn("Can't remove error cluster, skipping removal of " + name, e);
```

Comment:
nit: Error message could be clearer if it said something like: "Can't remove Dataproc Cluster <name>. Attempted deletion because state was ERROR"

Reply:
Done